src/Diagnostics.DataProviders/ChangeAnalysisClient.cs

```diff
@@ -0,0 +1,144 @@
+
+        /// <summary>
+        /// For detectors loaded from Diagnose and Solve, pass x-ms-client-principal-name to Change Analysis endpoint.
+        /// </summary>
+        private string clientPrincipalNameHeader;
+
+        private readonly string changeAnalysisEndPoint = "https://changeanalysis-dataplane-dev.azurewebsites.net/providers/microsoft.changeanalysis/";
+
+        private readonly Lazy<HttpClient> client = new Lazy<HttpClient>(() =>
+        {
+            var client = new HttpClient();
+            client.DefaultRequestHeaders.Accept.Clear();
+            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            return client;
+        }
+        );
+
+        private HttpClient httpClient
+        {
+            get
+            {
+                return client.Value;
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChangeAnalysisClient"/> class.
+        /// </summary>
+        public ChangeAnalysisClient(string clientObjectId, string clientPrincipalName = "")
+        {
+            clientObjectIdHeader = clientObjectId;
+            clientPrincipalNameHeader = clientPrincipalName;
+        }
+
+        /// <inheritdoc/>
+        public async Task<List<ResourceChangesResponseModel>> GetChangesAsync(ChangeRequest changeRequest)
+        {
+            string requestUri = changeAnalysisEndPoint + "changes?api-version=2019-04-01-preview";
+            object postBody = new
+            {
+                changeRequest.ResourceId,
+                changeRequest.ChangeSetId
+            };
+            string jsonString = await PrepareAndSendRequest(requestUri, postBody);
+            List<ResourceChangesResponseModel> resourceChangesResponse = JsonConvert.DeserializeObject<List<ResourceChangesResponseModel>>(jsonString);
+            return resourceChangesResponse;
+        }
+
+        /// <inheritdoc/>
+        public async Task<List<ChangeSetResponseModel>> GetChangeSetsAsync(ChangeSetsRequest changeSetsRequest)
+        {
+            string requestUri = changeAnalysisEndPoint + "changesets?api-version=2019-04-01-preview";
+            object postBody = new
+            {
+                changeSetsRequest.ResourceId,
+                StartTime = changeSetsRequest.StartTime.ToString(),
+                EndTime = changeSetsRequest.EndTime.ToString()
+            };
+            string jsonString = await PrepareAndSendRequest(requestUri, postBody);
+            List<ChangeSetResponseModel> changeSetsResponse = JsonConvert.DeserializeObject<List<ChangeSetResponseModel>>(jsonString);
+            return changeSetsResponse;
+        }
+
+        /// <inheritdoc/>
+        public async Task<ResourceIdResponseModel> GetResourceIdAsync(List<string> hostnames, string subscription)
+        {
+            string requestUri = changeAnalysisEndPoint + "resourceId?api-version=2019-04-01-preview";
```

Comment:
Have the api-version also defined in Config so that when change analysis team goes GA, we can simply update the api-version config.

Reply:
Sure, will do that.